inst/odin/carehomes.R

```diff
@@ -356,37 +363,84 @@ n_II_P[, , , ] <- n_I_P_progress[i, j, k, l] -
+##  R2 can progress tp E4 (w.r. strain 1 (3 - 2))
+rate_R_progress[, , ] <- waning_rate[i] +
+          rel_susceptibility[i, k] *
+            if (n_strains == 1 || j > 2 || model_super_infection == 0) 0 else
+              lambda[i, 3 - j]
+
+p_R_progress[, , ] <- 1 - exp(-rate_R_progress[i, j, k] * dt)
+
+## Total number who can possibly progress
+n_R_progress_tmp[, , ] <- rbinom(R[i, j, k], p_R_progress[i, j, k])
+
+
 ## cap on people who can move out of R based on numbers in T_sero_neg and
 ## T_PCR_neg
 n_R_progress_capped[, , ] <-
   min(n_R_progress_tmp[i, j, k], T_sero_neg[i, j, k], T_PCR_neg[i, j, k])
+
 ## use cap or not depending on model_pcr_and_serology value
+## n_R_progress is total number who either:
+##  - leave R for S or E and stay in same vacc class
+##  - leave R for S or E and change same vacc class
 n_R_progress[, , ] <- if (model_pcr_and_serology == 1)
   n_R_progress_capped[i, j, k] else n_R_progress_tmp[i, j, k]
-## of those some can also be vaccinated or progress through vaccination classes
-## --> number transitioning from R[j] to S[j+1]
-## (j vaccination class)
-n_RS_next_vacc_class[, , ] <-
-  rbinom(n_R_progress[i, j, k], p_R_next_vacc_class[i, j, k])
-## resulting transitions from R[j] to S[j]
-## (j vaccination class)
-n_RS[, , ] <- n_R_progress[i, j, k] - n_RS_next_vacc_class[i, j, k]
 
-## vaccine progression
-n_R_next_vacc_class_tmp[, , ] <- rbinom(
-  R[i, j, k] - n_R_progress[i, j, k], p_R_next_vacc_class[i, j, k])
-n_R_next_vacc_class_capped[, , ] <-
-  min(n_R_next_vacc_class_tmp[i, j, k],
+## Number going from R to S
+## In one-strain model, all progress to S only
+## In multi-strain model, R3 and R4 progress to S only
+## If not modelling super infection then all progress to S only
+##
+## In multi-strain model, number of R1 and R2 to S is binomial w.p. waning over
+##  waning plus prob strain
+## TODO (RS): waning_rate should eventually be variant varying
+p_RS[, , ] <- if (model_super_infection == 0 || n_strains == 1 || j > 2) 1 else
+                ((waning_rate[i] / (waning_rate[i] + lambda[i, 3 - j] *
+                                    rel_susceptibility[i, k])))
+n_RS[, , ] <- rbinom(n_R_progress[i, j, k], p_RS[i, j, k])
+
+
+## n_RE[i, j, k] is the number going from
+## R[age i, strain j, vacc class k] to
+## E[age i, strain j + 2, vacc class k]
+## Movement possible to E3 and E4 are from R1 and R2 respectively
+n_RE[, , ] <- n_R_progress[i, j, k] - n_RS[i, j, k]
+
+## Calculate those that change vacc class
+
+## 1) R -> S vaccine progression
+n_RS_next_vacc_class[, , ] <- if (n_vacc_classes == 1) 0 else
+  rbinom(n_RS[i, j, k], p_R_next_vacc_class[i, j, k])
+n_RS_same_vacc_class[, , ] <- n_RS[i, j, k] - n_RS_next_vacc_class[i, j, k]
+## 2) R -> E vaccine progression
+n_RE_next_vacc_class[, , ] <- if (n_vacc_classes == 1) 0 else
+  rbinom(n_RE[i, j, k], p_R_next_vacc_class[i, j, k])
+n_RE_same_vacc_class[, , ] <- n_RE[i, j, k] - n_RE_next_vacc_class[i, j, k]
+## 3) R -> R vaccine progression
```

Comment:
for all that section 3) R->R I would probably keep the code as it was: I don't think you need `n_RR` at all as we don't have sequential R compartments. So we just need to know how many people in R get vaccinated and stay in R, and that's exactly what we had before. So I don't think we need the extra variables `n_RR` and `n_RR_same_vacc_class`, and I think we can keep the simpler (and I think less misleading naming) of `n_R_next_vacc_class` rather than `n_RR_next_vacc_class`